### DISCUSSION TITLE
How to check if image is transparent

### DISCUSSION THREAD

**Discussion by SohamTilekar** [python-pillow/Pillow#7742] [Q&A] [answered]
How To Find Whether Pillow image is transparent or not.
- I am using this to pass in the paste method argument mask
---
Below is the My Solution.
- `img if img.mode[-1] in ('A', 'a') or (img.mode == 'P' and 'transparency' in img.info) else None`

Is there any more efficient way?

**Comment by radarhere** [comment, marked as answer]
Your code is [similar](https://github.com/python-pillow/Pillow/blob/1185fb82963ea77fe644e16baf0fcdc2a9b6b09a/src/PIL/Image.py#L1538-L1554) to `img.has_transparency_data` - so I imagine that is what you're looking for. It is an attribute that reports if an image has any transparency data or not.

See https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.has_transparency_data for documentation
